client/state/data-layer/wpcom-http/test/index.js

```diff
@@ -77,3 +80,99 @@ describe( '#queueRequest', () => {
 		}, 10 );
 	} );
 } );
+
+describe( '#fetcherMap', () => {
+	const wpcomReq = {};
+	const noopCallback = () => {};
+
+	describe( 'GET', () => {
+		let getFooAction;
+
+		beforeEach( () => {
+			wpcomReq.get = spy();
+			getFooAction = {
+				method: 'GET',
+				path: '/foo',
+			};
+		} );
+
+		it( 'should send with query', () => {
+			getFooAction.query = {
+				apiVersion: '2.0',
+				apiNamespace: 'wp/v2',
+			};
```

Comment:
this kind of mutation irks me because it hides the things that aren't here. interested at all in rewriting immutably?

```js
const action = { ...baseAction, query: { param: 'test' } };

// or since it's not too big we could envision just writing it out each time
const wpcom = {
	get: spy(),
	post: spy(),
};

beforeEach( () => wpcom.get.reset() );

const action = {
	method: 'GET',
	path: '/foo',
	query: { apiVersion: '1.3' }
}
```

we actually wrote fewer lines this way overall…